src/middleware/isFlaggedAndIsAdmin.js

```diff
@@ -0,0 +1,8 @@
+const isAdmin = require("./isAdmin")
```

Comment:
this works and is quite a cool approach. 

For our purposes though, I feel the title filename and function name don't capture the usage and will be hard to understand without reading the code.

I'm more biased toward us scoping this to the specific file. That is, to use it just where it's needed because it's quite specific. As it grows or the conditions change, we will possibly refactor it to any future use

So it will be similar code but directly in the `.get` route

```js
// This doesn't reflect the code we have as of now, just an idea
router()
.get(
  (req, res, next) => {
   // do if condition check
      // use isAdmin
  // else return next
}),
nextMiddleWareIfAny

.. rest of code
```

Reply:
yes please, thanks

Reply:
I have made all suggested changes, kindly review, thanks